pkg/cloudprovider/events/events.go

```diff
@@ -38,3 +42,13 @@ func NodeClaimFailedToResolveNodeClass(nodeClaim *v1beta1.NodeClaim) events.Even
 		DedupeValues:   []string{string(nodeClaim.UID)},
 	}
 }
+
+func FilteredInstanceTypes(nodeClaim *v1beta1.NodeClaim, resolvedNodeClassITs []string, filteredITs []string) events.Event {
+	return events.Event{
+		InvolvedObject: nodeClaim,
+		Type:           v1.EventTypeNormal,
+		Message: fmt.Sprintf("Resolved instance types from nodeclass: %s, filtered instance types after removing exotic types and spot instance types costlier than cheapest on-demand: %s", utils.PrettySlice(resolvedNodeClassITs, 5),
```

Comment:
Is it possible for us to create a message that shows how many instance types we filtered at each stage and then the last part of the message prints the remaining instance types using the pretty slice?

Reply:
@jonathan-innis Do you mean something like - `x instances were filtered/removed at a stage, y instances were filtered/removed at b stage, remaining instance types after filtering - u, v, w and 5 others`

Reply:
Also, just for reference, with the current changes the event looks like this - `Resolved instance types from nodeclass: r6in.metal, c6i.32xlarge, m6id.32xlarge, i4i.metal, m6in.32xlarge and 23 other(s), filtered instance types after removing exotic types and spot instance types costlier than cheapest on-demand: c6i.32xlarge, m6id.32xlarge, m6in.32xlarge, m6i.32xlarge, r6id.32xlarge and 9 other(s)`